cmd/machine-api-operator/client_builder.go

```diff
@@ -24,6 +25,11 @@ func (cb *ClientBuilder) OpenshiftClientOrDie(name string) osclientset.Interface
 	return osclientset.NewForConfigOrDie(rest.AddUserAgent(cb.config, name))
 }
 
+// MachineClientOrDie returns the machine api client interface for machine api objects.
+func (cb *ClientBuilder) MachineClientOrDie(name string) mapiclientset.Interface {
+	return mapiclientset.NewForConfigOrDie(rest.AddUserAgent(cb.config, name))
```

Comment:
I'm confused, why are you generating anything at all here?
Isn't all we need available here https://github.com/openshift/cluster-api/tree/openshift-4.2-cluster-api-0.1.0/pkg/client?
Can we just do exactly the same we do for the openshift config
mapiclientset "github.com/openshift/cluster-api/pkg/client/clientset_generated/clientset/typed/machine/v1beta1"

func (cb *ClientBuilder) KubeClientOrDie(name string) mapiclientset.MachinesGetter {
	c, _ := mapiclientset.NewForConfig(cb.config)
	return c
}

Reply:
vendored cluster-api at mao is having only clientset, no lister/informer. https://github.com/openshift/machine-api-operator/tree/master/vendor/github.com/openshift/cluster-api/pkg

I will see how to vendor missing parts as well, thanks!

Reply:
Resolved

Reply:
If you vendored the missing parts does https://github.com/openshift/machine-api-operator/pull/335#discussion_r300611726 still hold true as suggested?

Reply:
no, now informers and listers are being used as suggested.